test/onnx/test_pytorch_onnx_caffe2.py

```diff
@@ -882,13 +882,29 @@ def forward(self, x):
             self.run_model_test(MyModel(), train=False, input=(x), batch_size=BATCH_SIZE, use_gpu=False)
 
     # TODO: Add test cases for prod once Caffe2 has support for ReduceProd
-    def test_softmax(self):
+    def test_softmax_dim(self):
         for i in range(7)[2:]:
             model = nn.Softmax(dim=i - 1)
             dims = [2] * (i - 2) + [3, 4]
             input = torch.ones(*dims, requires_grad=True)
             self.run_model_test(model, train=False, batch_size=BATCH_SIZE, input=input)
 
+    def test_softmax(self):
```

Comment:
can we merge this test with the previous test (i.e., test_softmax_dim) or at least follow the style in the previous test. we can test the case which dim and axis have different semantics more thoroughly.

Reply:
I merged test_softmax_dim and test_softmax, but kept test_softmax_dtype separately, as we don't need to re-test each case with dtype